asm/f32/sdot.go

```diff
@@ -4,16 +4,16 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-package asm
+package f32
 
-func DsdotUnitary(x, y []float32) (sum float64) {
+func SdotUnitary(x, y []float32) (sum float64) {
```

Comment:
I think `f32.Ddot` is what we want (the `s` is already in the package selector); it's a `float32` routine that returns a `float64`. The alternative is `f32.Dotf64`, but that looks ugly.